### PR TITLE
[google_sign_in] Update (web) example app.

### PR DESCRIPTION
**Pull request by ditman**
## Changes

**google_sign_in:**

* Updates web bootstrap following [Flutter Web initialization docs](https://docs.flutter.dev/platform-integration/web/initialization#skipping-this-step).
* Uses the recent [`web_only` library](https://pub.dev/documentation/google_sign_in_web/0.12.3/web_only/web_only-library.html) to `renderButton`.
* Updates dependencies so it can compile with `--wasm`.

(opportunistic weekend commit) **google_identity_services_web:**

* Adds missing `CredentialSelectBy.fedcm_auto` enum value.
* Bump patch version + changelog.

## Issues

* Fixes https://github.com/flutter/flutter/issues/139852
* Missing bit of https://github.com/flutter/flutter/issues/138202

## Testing

* Deployed to: https://dit-gis-test.web.app

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by ditman** [comment]
I don't think this PR needs tests or a version, since nothing I changed is exposed through pub.dev

**Review by kevmoo — APPROVED** [review]
🙏

File: packages/google_sign_in/google_sign_in/example/lib/src/sign_in_button/web.dart [hunk 1]
```diff
@@ -3,13 +3,11 @@
...
-import 'package:google_sign_in_web/google_sign_in_web.dart' as web;
+import 'package:google_sign_in_web/web_only.dart' as web;
 
 import 'stub.dart';
 
 /// Renders a web-only SIGN IN button.
 Widget buildSignInButton({HandleSignInFn? onPressed}) {
-  return (GoogleSignInPlatform.instance as web.GoogleSignInPlugin)
-      .renderButton();
+  return web.renderButton();
```
Comment (ditman) [inline packages/google_sign_in/google_sign_in/example/lib/src/sign_in_button/web.dart:12]: I think this ended up looking much nicer than before @stuartmorgan!

**Comment by stuartmorgan** [comment]
test-exempt: code refactor with no semantic change
changelog override: example updates are minor, and don't need to be called out.

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**